test/Network/IpAddress.cpp

```diff
@@ -0,0 +1,223 @@
+#include <SFML/Network/IpAddress.hpp>
+
+#include <doctest.h>
+
+#include <sstream>
+
+using namespace std::string_literals;
+
+TEST_CASE("sf::IpAddress class - [network]")
+{
+    SUBCASE("Construction")
+    {
+        SUBCASE("Default constructor")
+        {
+            const sf::IpAddress ipAddress;
+            CHECK(ipAddress.toString() == "0.0.0.0"s);
+            CHECK(ipAddress.toInteger() == 0);
+            CHECK(ipAddress != sf::IpAddress::Any);
+        }
+
+        SUBCASE("std::string constructor")
+        {
+            const sf::IpAddress ipAddress = "192.168.0.1"s;
+            CHECK(ipAddress.toString() == "192.168.0.1"s);
+            CHECK(ipAddress.toInteger() == 0xC0A80001);
+            CHECK(ipAddress != sf::IpAddress::None);
+
+            const sf::IpAddress broadcast("255.255.255.255"s);
```

Comment:
Any reason why the previous constructor call is "implicit" (with `var = value` syntax) and this one "explicit" (with `var(value)` syntax)?

Reply:
`var(value)` is what all SFML tests tend to use by default but I added a few exceptional `var = value` uses to make sure we test the implicit-ness of these constructors.